PokeDemoTests/PokemonListPresenterTests.swift

```diff
@@ -0,0 +1,142 @@
+//
+//  PokeDemoTests.swift
+//  PokeDemoTests
+//
+//  Created by Rudr Bansal on 06/08/22.
+//
+
+import XCTest
+@testable import PokeDemo
+import Alamofire
+
+final class PokemonListPresenterTests: XCTestCase {
+    
+    var presenter: PokemonListPresenter!
+    
+    // MARK: - Lifecycle
+    
+    override func setUp() {
+        super.setUp()
+        presenter = PokemonListPresenter(service: Service.shared)
+    }
+    
+    override func tearDown() {
+        presenter = nil
+        super.tearDown()
+    }
+    
+    // MARK: - Tests
+    
+    func testDelegateSetsValidValue(){
+        // Given
+        let mockDelegate = MockPresenterDelegate()
+        
+        // When
+        presenter.setViewDelegate(delegate: mockDelegate)
+        
+        // Then
+        XCTAssertTrue(presenter.delegate is MockPresenterDelegate)
+    }
+    
+    func testGetPokemonListCallsAPI(){
+        // Given
+        let mockService = MockService()
+        
+        // When
+        presenter = PokemonListPresenter(service: mockService)
+        presenter.viewDidLoad()
+        
+        // Then
+        XCTAssertEqual(mockService.sendRequestWithJSONIsCalledIndex, 1)
+        XCTAssertEqual(mockService.sendRequestWithJSONEndPoint, "https://pokeapi.co/api/v2/pokemon")
+    }
```

Comment:
add asserts that right method, parameters, header are passed and completion is not nil